controllers/openstackmachine_controller.go

```diff
@@ -469,7 +470,16 @@ func (r *OpenStackMachineReconciler) getBootstrapData(ctx context.Context, machi
 		return "", errors.New("error retrieving bootstrap data: secret value key is missing")
 	}
 
-	return base64.StdEncoding.EncodeToString(value), nil
+	//return base64.StdEncoding.EncodeToString(value), nil
+	var buffer bytes.Buffer
+	buffer.Write(value)
+	if openStackMachine.Spec.AdminPass != ""{
+		adminPass := openStackMachine.Spec.AdminPass
+		resetpwd := "  - '(echo "+ adminPass +";sleep 1; echo "+ adminPass +") | passwd root' "
+		userPwd := []byte(resetpwd)
+		buffer.Write(userPwd)
+	}
+	return base64.StdEncoding.EncodeToString(buffer.Bytes()), nil
```

Comment:
I don't think this is the right place for this. I think instead you need to look into how to have your AdminPass added directly to the bootstrap data.

Reply:
exactly, I think this kind of info should be doable through openstack userdata etc?
add admin pwd here seems not a good place here

not sure https://docs.openstack.org/api-ref/compute/?expanded=change-administrative-password-changepassword-action-detail#change-administrative-password-changepassword-action
this is helpful ?